USFMToolsSharp/USFMParser.cs

```diff
@@ -38,22 +38,31 @@ public USFMDocument ParseFromString(string input)
                 ConvertToMarkerResult result = ConvertToMarker(match.Groups[1].Value, match.Groups[2].Value);
 
 
-                if(result.marker is TRMarker && !isInsideTable)
+                if(tempTable.TryInsert(result.marker))
                 {
-                    output.Insert(new TableBlock());
                     isInsideTable = true;
+
+                    if (!string.IsNullOrWhiteSpace(result.remainingText))
+                    {
+                        tempTable.TryInsert(new TextBlock(result.remainingText));

```

Comment:
What happens if the TryInsert fails?  It looks like the remaining text just gets dropped?

Reply:
@PurpleGuitar I wasn't super confident of how to handle the misplaced TextBlock. So the updates I've made to the parser adds all Un-Insertable TextBlocks into the document before the table is inserted.

Reply:
I can live with that.  LGTM.